File: src/main/java/com/devmpv/model/Projections.java
package com.devmpv.model;

import org.springframework.data.rest.core.config.Projection;

import java.util.Set;

public interface Projections {

    @Projection(name = "inlineAttachments", types = {Thread.class, Message.class})
    interface InlineAttachments {

        Set<Attachment> getAttachments();

        Set<Long> getReplyIds();

        long getId();

        String getText();

        long getCreatedAt();

        String getTitle();

        long getUpdated();
    }
}

File: src/main/kotlin/com/devmpv/model/Projections.kt
package com.devmpv.model

import org.springframework.data.rest.core.config.Projection

interface Projections {

    @Projection(name = "inlineAttachments", types = arrayOf(Thread::class, Message::class))
    interface InlineAttachments {

        val attachments: Set<Attachment>

        val replyIds: Set<Long>

        val id: Long

        val text: String

        val createdAt: Long

        val updatedAt: Long

        val title: String
    }
}
